src/PointwiseFunctions/GeneralRelativity/GeneralizedHarmonic/CovariantDerivOfExtrinsicCurvature.cpp

```diff
@@ -0,0 +1,136 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/GeneralRelativity/GeneralizedHarmonic/CovariantDerivOfExtrinsicCurvature.hpp"
+
+#include "DataStructures/DataVector.hpp"     // IWYU pragma: keep
+#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+// IWYU pragma: no_forward_declare Tensor
+
+/// \cond
+namespace GeneralizedHarmonic {
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::ijj<DataType, SpatialDim, Frame> covariant_deriv_of_extrinsic_curvature(
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal_vector,
+    const tnsr::Ijj<DataType, SpatialDim, Frame>&
+        spatial_christoffel_second_kind,
+    const tnsr::ii<DataType, SpatialDim, Frame>& extrinsic_curvature,
+    const tnsr::AA<DataType, SpatialDim, Frame>& inverse_spacetime_metric,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& d_pi,
+    const tnsr::ijaa<DataType, SpatialDim, Frame>& d_phi) noexcept {
+  tnsr::ijj<DataType, SpatialDim, Frame> d_extrinsic_curvature{};
+  GeneralizedHarmonic::covariant_deriv_of_extrinsic_curvature<SpatialDim, Frame,
+                                                              DataType>(
+      make_not_null(&d_extrinsic_curvature), spacetime_unit_normal_vector,
+      spatial_christoffel_second_kind, extrinsic_curvature,
+      inverse_spacetime_metric, phi, d_pi, d_phi);
+  return d_extrinsic_curvature;
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+void covariant_deriv_of_extrinsic_curvature(
+    const gsl::not_null<tnsr::ijj<DataType, SpatialDim, Frame>*>
+        d_extrinsic_curvature,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal_vector,
+    const tnsr::Ijj<DataType, SpatialDim, Frame>&
+        spatial_christoffel_second_kind,
+    const tnsr::ii<DataType, SpatialDim, Frame>& extrinsic_curvature,
+    const tnsr::AA<DataType, SpatialDim, Frame>& inverse_spacetime_metric,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& d_pi,
+    const tnsr::ijaa<DataType, SpatialDim, Frame>& d_phi) noexcept {
+  if (UNLIKELY(get_size(get<0, 0, 0>(*d_extrinsic_curvature)) !=
+               get_size(get<0>(spacetime_unit_normal_vector)))) {
+    *d_extrinsic_curvature = tnsr::ijj<DataType, SpatialDim, Frame>(
+        get_size(get<0>(spacetime_unit_normal_vector)));
+  }
+
+  // Ordinary derivative first
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    for (size_t j = i; j < SpatialDim; ++j) {
+      for (size_t k = 0; k < SpatialDim; ++k) {
+        d_extrinsic_curvature->get(k, i, j) = 0.5 * d_pi.get(k, i + 1, j + 1);
```

Comment:
Since every term has 0.5 in it, why not do that multiplication at the end?

Reply:
good point, updated the code to multiply by half while adding in the Christoffel symbol terms at the end.

Reply:
good point, updated the code to do so right at the end while adding in the terms with Christoffel symbols.